### ISSUE TITLE
Inconsistency for the convolutional layer between Tensorflow and Theano backends

### ISSUE THREAD

**Issue by Dandy-John** [keras-team/keras#12843]
### Configuration

| Library                        | Version  |
| ------------------------------ | -------- |
| python                         | 3.6.8    |
| GCC                            | 7.3.0    |
| tensorflow-base/tensorflow-gpu | 1.13.1   |
| keras-gpu/keras-base           | 2.2.4    |
| theano                         | 1.0.4    |
| cudnn                          | 7.3.1    |
| cudatoolkit                    | 10.0.130 |

Machine Configuration：

Ubuntu 18.04.2 LTS

Intel(R) Xeon(R) CPU E5-2640 v4 @ 2.40GHz 128GB RAM

CUDA Version: 10.2 

Double GeForce GTX 1080ti 

---

### Description

I used the same pre-trained keras model (a LeNet structure on fashion-mnist datasets) on tensorflow and theano backends respectively, and noticed that there is a big difference between the two backends. The accuracy on tensorflow is 91.6% while the accuracy on theano is just 40.6%.

![acc](https://user-images.githubusercontent.com/17698785/58078171-abfebd00-7be0-11e9-8980-a3e2686cac16.png)

The prediction results on the two backends are shown as below, respectively:

![tensorflow-result](https://user-images.githubusercontent.com/17698785/58078197-bcaf3300-7be0-11e9-91a5-2ad46142c1ef.png)
![theano-result](https://user-images.githubusercontent.com/17698785/58078200-bcaf3300-7be0-11e9-83f3-4ed88c93991d.png)

Based on the results, the problem seems to occur at the backend of Theano. I also tried to localize the problem, and found that the inconsistency may happen at the first layer of the model, which is a convolutional layer:

```python
keras.layers.Conv2D(filters=64, kernel_size=2, padding='same', activation='relu', input_shape=(28, 28, 1))
```
I compared the outputs of the layer from the two backends, and found that many values are not close (or the same).

This layer is also used in many other models but there is no such inconsistency. Therefore, I suspect this problem happens under some specific context (such as parameters). In particular, I constructed a model only with this layer (using the same parameters and shuffling the weights), and the inconsistency still exists.

I attched the code and the pretrained model (.h5 file) to help reproduce this inconsistency.

[inconsistency_1.zip](https://github.com/keras-team/keras/files/3201344/inconsistency_1.zip)
